zerver/lib/events.py

```diff
@@ -1024,6 +1024,8 @@ def apply_event(
         if "raw_unread_msgs" in state and event["flag"] == "read" and event["op"] == "add":
             for remove_id in event["messages"]:
                 remove_message_id_from_unread_mgs(state["raw_unread_msgs"], remove_id)
+        if event["flag"] == "read" and event["op"] == "remove":
+            state["raw_unread_msgs"] = get_raw_unread_data(user_profile, event["unread_data"])
```

Comment:
This is an expensive function call; we should at least document that in a comment. But in an ideal world we'd actually got through and put values where they belong.

Reply:
@timabbott I'm not sure I understand that? You mean that we should be able to have the raw_unread_data already in the event dictionary already? We would have to anyway perform this operation at some point, right?